src/bundle/Command/AbstractMultiProcessComand.php

```diff
@@ -0,0 +1,356 @@
+        PermissionResolver $permissionResolver,
+        UserService $userService
+    ) {
+        $this->permissionResolver = $permissionResolver;
+        $this->userService = $userService;
+        $this->dryRun = false;
+        $this->processes = [];
+
+        parent::__construct($name);
+    }
+
+    public function configure(): void
+    {
+        $this->
+            addOption(
+                'user',
+                'u',
+                InputOption::VALUE_REQUIRED,
+                'Ibexa DXP username',
+                'admin'
+            )->addOption(
+                'dry-run',
+                null,
+                InputOption::VALUE_NONE,
+                'Run the converter without writing anything to the database'
+            )->addOption(
+                'no-progress',
+                null,
+                InputOption::VALUE_NONE,
+                'Disable the progress bar.'
+            )->addOption(
+                'processes',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Number of child processes to run in parallel for iterations, if set to "auto" it will set to number of CPU cores -1, set to "1" or "0" to disable [default: "auto"]',
+                1
+            )->addOption(
+                'iteration-count',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Number of objects to process in a single iteration. Set to avoid using too much memory [default: 10000]',
+                10000
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->user = (string) $input->getOption('user');
+        $this->permissionResolver->setCurrentUserReference(
+            $this->userService->loadUserByLogin($this->user)
+        );
+
+        $this->environment = (string) $input->getOption('env');
+
+        if ($input->getOption('dry-run')) {
+            $this->dryRun = true;
+        }
+        if ($this->isDryRun() && !$this->isChildProcess()) {
+            $output->writeln('Running in dry-run mode. No changes will actually be written to database');
+        }
+
+        $this->maxProcesses = (int) $input->getOption('processes');
+        if ($this->maxProcesses < 1) {
+            throw new RuntimeException('Invalid value for "--processes" given');
+        }
+
+        $this->iterationCount = (int) $input->getOption('iteration-count');
+        if ($this->iterationCount < 1) {
+            throw new RuntimeException('Invalid value for "--processes" given');
+        }
+
+        $this->hasProgressBar = !$this->isChildProcess() && !$input->getOption('no-progress');
+
+        $this->output = $output;
+
+        if ($this->isChildProcess()) {
+            $cursor = $this->constructCursorFromInputOptions();
+            $this->processData($cursor);
+        } else {
+            $this->output->writeln('Processing ' . $this->getObjectCount() . ' items.');
```

Comment:
When the query to count the affected number of rows can take a long time (in our case more than 10 minutes), there is no output because this would be the first line. Please change that because it no feedback for multiple minutes can cause major confusion about the process.

Reply:
Fixed in 5abb5f763e76b0d5e0b6c655e03bd8da1ea44888